### PR TITLE
Add frame.h

### PR DESCRIPTION
**Pull request by choucl**
I have add some opcode and status definitions in `frame.h` which is used for socket.
The definitions are according to the following diagram.

image

### CODE REVIEW

**Comment by choucl** [comment]
Please let me know if there's anything need to be modified😄

**Comment by choucl** [comment]
I have changed the original *REQ_DROP* opcode into *REQ_PLACE*

File: inc/frame.h [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    PAIRED,
+    REQ_TURN,
+    REQ_STATUS,
+    RES,
+    REQ_PLACE
+};
+
+#define STATUS_INV -1
+#define STATUS_OK 1
+#define STATUS_FAILED 0
```
Comment (yuyuranium) [inline inc/frame.h:20]: Are we going to use the lower 8 bits as the `status` code?

Comment (choucl) [reply inc/frame.h:20]: Yes, the lower 8 bits will be used as `status` code.

The status codes you provided seem great, I'll change the code to this version

Comment (yuyuranium) [reply inc/frame.h]: Great, thanks!

File: inc/frame.h [hunk 2]
```diff
@@ -15,8 +15,8 @@ enum OPCODE {
     REQ_PLACE
 };
 
-#define STATUS_INV -1
-#define STATUS_OK 1
-#define STATUS_FAILED 0
+#define OK 0
+#define INV -1
+#define BAD_REQ -2
```
Comment (yuyuranium) [inline inc/frame.h:20]: You are right. Conflict occurs. 🥲

Comment (yuyuranium) [reply inc/frame.h:20]: I think this is more like a **response code**. Maybe we can name these definitions using this concept.

Comment (choucl) [reply inc/frame.h:20]: How about adding a prefix of *RES_*?

**Review by yuyuranium — APPROVED** [review]
LGTM!

**Pull request merged**